classes/class-sensei-media-attachments.php

```diff
@@ -183,7 +183,7 @@ public function display_attached_media() {
 
 		$user_id    = get_current_user_id();
 		$course_id  = 'course' === $post_type ? $post->ID : get_post_meta( $post->ID, '_lesson_course', true );
-		$show_links = Sensei_Utils::user_started_course( $course_id, $user_id );
+		$show_links = Sensei_Utils::user_started_course( $course_id, $user_id ) || current_user_can( 'edit_courses' );
```

Comment:
Adding this cap means teachers will be able to view media attachments for courses they are not assigned to. They should only be able to view media for courses where they are the assigned teacher.